core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -190,11 +218,26 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
     };
 
     Topic.prototype.addUncategorizedSkillId = function(skillId) {
+      var skillIsPresentInSomeSubtopic = false;
+      for (var i = 0; i < this._subtopics.length; i++) {
+        var skillIds = this._subtopics[i].getSkillIds();
+        if (skillIds.indexOf(skillId) !== -1) {
+          skillIsPresentInSomeSubtopic = true;
+          break;
+        }
+      }
+      if (this._uncategorizedSkillIds.indexOf(skillId) !== -1 ||
+          skillIsPresentInSomeSubtopic) {
+        throw Error('Given skillId is already present in a subtopic.');
```

Comment:
Well, not if it's in uncategorizedSkillIds, right?

Reply:
Done, split the check to throw different errors.